test/end2end_test.go

```diff
@@ -613,9 +613,9 @@ func testServerGoAway(t *testing.T, e env) {
 		}
 		break
 	}
-	// A new RPC should fail with Unavailable error.
-	if _, err := tc.EmptyCall(context.Background(), &testpb.Empty{}); err == nil || grpc.Code(err) != codes.Unavailable {
-		t.Fatalf("TestService/EmptyCall(_, _) = _, %v, want _, error code: %d", err, codes.Unavailable)
+	// A new RPC should fail.
+	if _, err := tc.EmptyCall(context.Background(), &testpb.Empty{}); err == nil {
```

Comment:
I'm always concerned about checks like this. Can you check for a specific error (or errors) rather than just any non-nil?